### DISCUSSION TITLE
Different input data layout

### DISCUSSION THREAD

**Discussion by haavardf** [eprbell/rp2#129] [Ideas]
Hi, and thank you for providing RP2.

I found the input format a bit difficult to use, and have a suggestion for what I think might be an improvement. I am aware that there would be lot's of users out there with old data, so of course this must be a choise to use.

May main point when using the current input format is:
- Tables IN, INTRA and OUT is on the same sheet, which colums spaning different type of data.
- There is one sheet for each crypto, I keep my data as one long record, where I note the type of currency.
- The ability to have other sheets, like my wallet-sheet (see screenshot) where I keep track of my addresses and automatically can fill in owner and exchange/wallet depending on the address, which I keep in the transaction records.

![Screenshot_20240803_075637](https://github.com/user-attachments/assets/5e181555-ba2e-468f-8589-defdbf0ed924)

I have merged all crypto currency into 3 sheets, IN, INTRA and OUT. The picture is probably the best explanation.

I have a working copy of this, so if this is something that might be helpfull for the project or others let me know how best to proceed.

I don't want to step on anyone, so please be frank.

**Comment by eprbell** [comment]
For format conversion, the recommended way is to write a [DaLI data loader plugin](https://github.com/eprbell/dali-rp2/blob/main/README.dev.md#data-loader-plugin-development), which can read your format. Then DaLI automatically takes care of generating RP2-compliant input files. As boilerplate you can use the [manual plugin](https://github.com/eprbell/dali-rp2/blob/main/docs/configuration_file.md#manual-section-csv), which does something similar to what you need (see also [the code](https://github.com/eprbell/dali-rp2/blob/main/src/dali/plugin/input/csv/manual.py)).
